validator/client/src/main/java/tech/pegasys/teku/validator/client/duties/synccommittee/ObolDvtSyncAggregatorSelectionProofProvider.java

```diff
@@ -0,0 +1,130 @@
+      final UInt64 slot,
+      final Collection<ValidatorAndCommitteeIndices> assignments,
+      final SyncCommitteeUtil syncCommitteeUtil,
+      final ForkInfo forkInfo) {
+
+    final Set<SafeFuture<SyncCommitteeSelectionProof>> partialSelectionProofFutures =
+        new HashSet<>();
+
+    assignments.forEach(
+        assignment -> {
+          final IntStream syncSubCommitteeIndexes =
+              syncCommitteeUtil.getSyncSubcommittees(assignment.getCommitteeIndices()).intStream();
+
+          syncSubCommitteeIndexes.forEach(
+              subcommitteeIndex -> {
+                selectionProofFutures.put(
+                    Pair.of(assignment.getValidatorIndex(), subcommitteeIndex), new SafeFuture<>());
+
+                final SafeFuture<SyncCommitteeSelectionProof> partialProofFuture =
+                    createAndSignSelectionData(
+                            slot, syncCommitteeUtil, forkInfo, assignment, subcommitteeIndex)
+                        .thenApply(
+                            partialProof ->
+                                mapToSelectionProofRequest(
+                                    partialProof,
+                                    slot,
+                                    assignment.getValidatorIndex(),
+                                    subcommitteeIndex));
+
+                partialSelectionProofFutures.add(partialProofFuture);
+              });
+        });
+
+    return SafeFuture.allOfFailFast(partialSelectionProofFutures.toArray(new SafeFuture[0]))
+        .thenCompose(
+            __ -> {
+              final List<SyncCommitteeSelectionProof> requests =
+                  partialSelectionProofFutures.stream().map(SafeFuture::getImmediately).toList();
+
+              return validatorApiChannel
+                  .getSyncCommitteeSelectionProof(requests)
+                  .thenCompose(
+                      combinedSelectionProofs ->
+                          combinedSelectionProofs
+                              .map(this::processCombinedSelectionProofsResponse)
+                              .orElseGet(
+                                  () ->
+                                      SafeFuture.failedFuture(
+                                          new RuntimeException(
+                                              "Empty response from Obol DVT integration"))));
+            });
+  }
+
+  private SafeFuture<Void> processCombinedSelectionProofsResponse(
+      final List<SyncCommitteeSelectionProof> combinedSelectionProofs) {
+    combinedSelectionProofs.forEach(
+        combinedProof -> {
+          final Pair<Integer, Integer> key =
+              Pair.of(combinedProof.getValidatorIndex(), combinedProof.getSubcommitteeIndex());
+
+          if (selectionProofFutures.containsKey(key)) {
+            selectionProofFutures.get(key).complete(combinedProof.getSelectionProofSignature());
+          } else {
+            throw new RuntimeException(
```

Comment:
i think maybe rather than spamming RuntimeException we create an `ObolIntegrationException` or something?